packages/forwarder-helper/src/forwarder_helper_factory.ts

```diff
@@ -22,4 +31,220 @@ export const forwarderHelperFactory = {
+            );
+            // take fee orders from SRA + on chain information and find the valid orders and remaining fillable maker asset amounts
+            feeOrdersAndRemainingFillableMakerAssetAmounts = getValidOrdersAndRemainingFillableMakerAssetAmountsFromOnChain(
+                feeOrdersFromSra,
+                feeOrdersAndTradersInfo,
+                zrxTokenAssetData,
+            );
+        } else {
+            // if we don't have an rpc url, assume all orders are valid and fallback to optional fill amounts from SRA
+            // if fill amounts are not available from the SRA, assume all orders are completely fillable
+            const apiOrdersFromSra = makerAssetOrderbook.asks.records;
+            const feeApiOrdersFromSra = zrxOrderbook.asks.records;
+            // take maker asset orders from SRA and the valid orders and remaining fillable maker asset amounts
+            ordersAndRemainingFillableMakerAssetAmounts = getValidOrdersAndRemainingFillableMakerAssetAmountsFromApi(
+                apiOrdersFromSra,
+            );
+            // take fee orders from SRA and find the valid orders and remaining fillable maker asset amounts
+            feeOrdersAndRemainingFillableMakerAssetAmounts = getValidOrdersAndRemainingFillableMakerAssetAmountsFromApi(
+                feeApiOrdersFromSra,
+            );
+        }
+        // compile final config
+        const config: ForwarderHelperImplConfig = {
+            orders: ordersAndRemainingFillableMakerAssetAmounts.orders,
+            feeOrders: feeOrdersAndRemainingFillableMakerAssetAmounts.orders,
+            remainingFillableMakerAssetAmounts:
+                ordersAndRemainingFillableMakerAssetAmounts.remainingFillableMakerAssetAmounts,
+            remainingFillableFeeAmounts:
+                feeOrdersAndRemainingFillableMakerAssetAmounts.remainingFillableMakerAssetAmounts,
+        };
+        const helper = new ForwarderHelperImpl(config);
+        return helper;
+    },
 };
+
+interface OrdersAndRemainingFillableMakerAssetAmounts {
+    orders: SignedOrder[];
+    remainingFillableMakerAssetAmounts: BigNumber[];
+}
+
+/**
+ * Given an array of APIOrder objects from a standard relayer api, return an array
+ * of fillable orders with their corresponding remainingFillableMakerAssetAmounts
+ */
+function getValidOrdersAndRemainingFillableMakerAssetAmountsFromApi(
+    apiOrders: APIOrder[],
+): OrdersAndRemainingFillableMakerAssetAmounts {
+    const result = _.reduce(
+        apiOrders,
+        (acc, apiOrder, index) => {
+            // get current accumulations
+            const { orders, remainingFillableMakerAssetAmounts } = acc;
+            // get order and metadata
+            const { order, metaData } = apiOrder;
+            // if the order is expired, move on
+            if (orderUtils.isOrderExpired(order)) {
+                return acc;
+            }
+            // calculate remainingFillableMakerAssetAmount from api metadata
```

Comment:
add comment about trying to grab remaining taker amount from SRA metadata, else, assume order is completely fillable